### PR TITLE
add v1 to hack/test-go; let default value of Container.Security Context to be nil

### PR DESCRIPTION
**Pull request by caesarxuchao**
enable unit tests for v1; let default value of Container.Security Context to be nil
#7343 #9070 #7018

I made a separate PR #9088 for the v1 swagger-spec changes. I will rebase this PR on #9088 later.

@bgrant0607 @krousey @nikhiljindal @pmorie

### CODE REVIEW

File: pkg/kubelet/config/http_test.go [hunk 1]
```diff
@@ -161,7 +160,8 @@ func TestExtractPodsFromHTTP(t *testing.T) {
 							Image: "foo",
 							TerminationMessagePath: "/dev/termination-log",
 							ImagePullPolicy:        "Always",
-							SecurityContext:        securitycontext.ValidSecurityContextWithContainerDefaults()}},
+							//SecurityContext:        securitycontext.ValidSecurityContextWithContainerDefaults()
```
Comment (bgrant0607) [inline pkg/kubelet/config/http_test.go]: Just delete

File: pkg/api/validation/testdata/v1/invalidPod.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    name: redis-master
+  name: name
+spec:
+  containers:
+  - args: "this is a bad command"
```
Comment (bgrant0607) [inline pkg/api/validation/testdata/v1/invalidPod.yaml]: Please reorder the container fields to match the json: name, image, args.

**Comment by bgrant0607** [comment]
Just a nit.

**Comment by caesarxuchao** [comment]
@bgrant0607 PTAL. Thanks.

Do you still want me to reorder the fields in the pod file?

**Comment by bgrant0607** [comment]
@caesarxuchao Don't bother. I don't care as much for tests as for examples.

**Comment by bgrant0607** [comment]
LGTM

File: pkg/api/latest/latest_test.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ import (
 	"testing"
 
 	internal "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	_ "github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1"
```
Comment (bgrant0607) [inline pkg/api/latest/latest_test.go]: Please remove this. The subsequent PRs include this also.

Comment (caesarxuchao) [inline pkg/api/latest/latest_test.go]: Yes, I will.

Comment (bgrant0607) [inline pkg/api/latest/latest_test.go]: ping. All your PRs are blocked on this.

**Comment by caesarxuchao** [comment]
@bgrant0607, I remove the import of v1 package in all PRs. I will rebase and squash them in the order of  #9101 #9145 #9154 #9169

**Comment by caesarxuchao** [comment]
This one is rebased and squashed.

**Comment by bgrant0607** [comment]
Rerunning Shippable.

**Pull request merged**